packages/server/graphql/mutations/resetRetroMeetingToGroupStage.ts

```diff
@@ -0,0 +1,135 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import {CHECKIN, DISCUSS, GROUP, REFLECT, VOTE} from '../../../client/utils/constants'
+import getRethink from '../../database/rethinkDriver'
+import DiscussPhase from '../../database/types/DiscussPhase'
+import GenericMeetingPhase from '../../database/types/GenericMeetingPhase'
+import MeetingRetrospective from '../../database/types/MeetingRetrospective'
+import {getUserId} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import standardError from '../../utils/standardError'
+import ResetRetroMeetingToGroupStagePayload from '../types/ResetRetroMeetingToGroupStagePayload'
+import {primePhases} from './helpers/createNewMeetingPhases'
+
+const resetRetroMeetingToGroupStage = {
+  type: GraphQLNonNull(ResetRetroMeetingToGroupStagePayload),
+  description: `Reset a retro meeting to group stage`,
+  args: {
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID)
+    }
+  },
+  resolve: async (_source, {meetingId}, {authToken, socketId: mutatorId, dataLoader}) => {
+    const r = await getRethink()
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+
+    // AUTH
+    const viewerId = getUserId(authToken)
+    const meeting = (await dataLoader.get('newMeetings').load(meetingId)) as MeetingRetrospective
+    if (!meeting) return standardError(new Error('Meeting not found'), {userId: viewerId})
+    const {createdBy, facilitatorUserId, phases, meetingType} = meeting
+    if (meetingType != 'retrospective') {
+      return standardError(new Error('Meeting type is not retrospective'), {userId: viewerId})
+    }
+    if (viewerId !== facilitatorUserId) {
+      if (viewerId !== createdBy)
+        return standardError(new Error('Not meeting facilitator'), {userId: viewerId})
+      return standardError(new Error('Not meeting facilitator anymore'), {userId: viewerId})
+    }
+
+    // VALIDATION
+    const groupPhase = phases.find((phase) => phase.phaseType === GROUP)
+    if (!groupPhase) return standardError(new Error('Group phase not found'), {userId: viewerId})
+    const resetToStage = groupPhase.stages.find((stage) => stage.phaseType === GROUP)
+    if (!resetToStage) return standardError(new Error('Group stage not found'), {userId: viewerId})
+    if (!resetToStage.isNavigableByFacilitator)
+      return standardError(new Error('Group stage has not started'), {userId: viewerId})
+    if (!resetToStage.isComplete)
+      return standardError(new Error('Group stage has not finished'), {userId: viewerId})
+    if (meeting.endedAt)
+      return standardError(new Error('The meeting has already ended'), {userId: viewerId})
+
+    // RESOLUTION
+    const discussionPhase = phases.find((phase) => phase.phaseType === DISCUSS) as DiscussPhase
```

Comment:
@nickoferrall built a great little helper functrion so we no longer have to cast! https://github.com/ParabolInc/parabol/blob/8118227c74f0df6e3da4146bec61dab3ddc02f4e/packages/server/utils/getPhase.ts